src/simulation/dynamics/spinningBodies/spinningBodiesOneDOF/_UnitTest/test_spinningBodyOneDOFStateEffector.py

```diff
@@ -296,5 +296,141 @@ def spinningBody(show_plots, cmdTorque, lock, thetaRef):
+    unitProcessName = "TestProcess"  # arbitrary name (don't change)
+
+    #   Create a sim module as an empty container
+    unitTestSim = SimulationBaseClass.SimBaseClass()
+
+    # Create test thread
+    testProcessRate = macros.sec2nano(0.1)  # update process rate update time
+    testProc = unitTestSim.CreateNewProcess(unitProcessName)
+    testProc.addTask(unitTestSim.CreateNewTask(unitTaskName, testProcessRate))
+
+    # Create the spacecraft module
+    scObject = spacecraft.Spacecraft()
+    scObject.ModelTag = "spacecraftBody"
+
+    # Define mass properties of the rigid hub of the spacecraft
+    scObject.hub.mHub = 750.0
+    scObject.hub.r_BcB_B = [0.0, 0.0, 0.0]
+    scObject.hub.IHubPntBc_B = [[900.0, 0.0, 0.0], [0.0, 800.0, 0.0], [0.0, 0.0, 600.0]]
+    unitTestSim.AddModelToTask(unitTaskName, scObject)
+
+    # Create a spinning body
+    spinningBody = spinningBodyOneDOFStateEffector.SpinningBodyOneDOFStateEffector()
+    spinningBody.mass = 50.0
+    spinningBody.IPntSc_S = [[50.0, 0.0, 0.0], [0.0, 30.0, 0.0], [0.0, 0.0, 40.0]]
+    spinningBody.dcm_S0B = [[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 1.0]]
+    spinningBody.r_ScS_S = [[1.0], [0.0], [-1.0]]
+    spinningBody.r_SB_B = [[0.5], [-1.5], [-0.5]]
+    spinningBody.sHat_S = [[0], [-1], [0]]
+    spinningBody.theta_max = np.deg2rad(max_lim)
+    spinningBody.theta_min = np.deg2rad(min_lim)
+    spinningBody.ModelTag = "SpinningBody"
+    unitTestSim.AddModelToTask(unitTaskName, spinningBody)
+    scObject.addStateEffector(spinningBody)
+
+    # Create the torque message
+    cmdArray = messaging.ArrayMotorTorqueMsgPayload()
+    cmdArray.motorTorque = [1] + [0] * 5  # [Nm]
+    cmdMsg = messaging.ArrayMotorTorqueMsg().write(cmdArray)
+    spinningBody.motorTorqueInMsg.subscribeTo(cmdMsg)
+
+    # Initialize the simulation
+    unitTestSim.InitializeSimulation()
+
+    # Setup logging
+    spinning_body_log = spinningBody.spinningBodyOutMsg.recorder()
+    unitTestSim.AddModelToTask(unitTaskName, spinning_body_log)
+    sc_log = scObject.scStateOutMsg.recorder()
+    unitTestSim.AddModelToTask(unitTaskName, sc_log)
+    torque_log = cmdMsg.recorder()
+    unitTestSim.AddModelToTask(unitTaskName, torque_log)
+
+    # Setup and run the simulation
+    stopTime = 200 * testProcessRate
+    unitTestSim.ConfigureStopTime(stopTime)
+    unitTestSim.ExecuteSimulation()
+
+    # run with opposite torque
+    cmdArray.motorTorque = [-1] + [0] * 5  # [Nm]
+    cmdMsg = cmdMsg.write(cmdArray)
+    scObject.hub.omega_BN_BInit = [0.0, 0.0, 0.0]
+
+    unitTestSim.ConfigureStopTime(3 * stopTime)  # run twice as long to ensure body reaches opposite side
+    unitTestSim.ExecuteSimulation()
+
+    # Extract the logged variables
+    time_sec = np.array(sc_log.times()) * macros.NANO2SEC
+    theta = spinning_body_log.theta
+    torque = torque_log.motorTorque[:, 0]
+    thetaDot = spinning_body_log.thetaDot
+    omega = sc_log.omega_BN_B
+
```

Comment:
please close all prior plots before making new plots.  This is helpful when running lots of tests at once.